### ISSUE TITLE
learnyoureact : Please provide the solution for Isomorphic problem (problem no 8). 

### ISSUE THREAD

**Issue by vineetasharma** [nodeschool/discussions#1937]
I have tried to solve the problem but it is  not meeting the acceptance criteria .

* node version: 6.3.0
* npm version:  3.10.3
* os: linux

### output
```
 // Its working fine over server.
```

### My Code
```js
 //app.js

import React from 'react';
import ReactDOM from 'react-dom';
import TodoBox from './views/view7.jsx';

console.log('App.js loaded--------------------');
let data = JSON.parse(document.getElementById('initial-data').getAttribute('data-json'));
ReactDOM.render(<TodoBox data={data} />, document.getElementById("app"));
```

```js
// program.js

var React = require('react');
var ReactDOMServer = require('react-dom/server');
var DOM = React.DOM;
var body = DOM.body;
var div = DOM.div;
var script = DOM.script;

var browserify = require('browserify');
var babelify = require("babelify");


var express = require('express');
var app = express();

app.set('port', (process.argv[2] || 3000));
app.set('view engine', 'jsx');
app.set('views', __dirname + '/views');
app.engine('jsx', require('express-react-views').createEngine({ transformViews: false }));

require('babel/register')({
    ignore: false
});

var TodoBox = require('./views/view7.jsx');




app.use('/bundle.js', function (req, res) {
    res.setHeader('content-type', 'application/javascript');

    browserify({ debug: true })
        .transform(babelify.configure({
            presets: ["react", "es2015"],
            compact: false
        }))
        .require("./app.js", { entry: true })
        .bundle()
        .pipe(res);
});


var data = [{title:"Shopping",detail:process.argv[3]},{title:"Hair cut",detail:process.argv[4]}];

app.use('/', function (req, res) {
    var initialData = JSON.stringify(data);
    var markup = ReactDOMServer.renderToString(React.createElement(TodoBox, {data: data}));


    res.setHeader('Content-Type', 'text/html');

    var html = ReactDOMServer.renderToStaticMarkup(body(null,
        div({id: 'app', dangerouslySetInnerHTML: {__html: markup}}),
        script({
            id: 'initial-data',
            type: 'text/plain',
            'data-json': initialData
        }),
        script({src: '/bundle.js'})
    ));

    console.log('Markup----------------------------------------',html);

    res.end(html);
});

app.listen(app.get('port'), function() {});

```

```js
// view7.jsx
import React from 'react';

export default class TodoBox extends React.Component {
    render() {
        return (
            <div className="todoBox">
                <h1>Todos</h1>
                <div className="todoList">
                <TodoList data = {this.props.data} />
                </div>
                <TodoForm />
            </div>
        );
    }
}

class TodoList extends React.Component {
    render() {
        var todo = this.props.data.map(function(obj) { return <Todo title={obj.title} key={obj.title}>{obj.detail}</Todo>});
        return (
                <table style={{border: "2px solid black"}} >
                    {todo}
                </table>
        );
    }
}


class TodoForm extends React.Component {
    // Write code here
    render(){
        return (
            <div className="todoForm">
            I am a TodoForm.
        </div>
        );
    }
}


class Todo extends React.Component {
    constructor(props){
        super(props);
        this.state = {checked: false};
        this.handleChange=function(event) {
            this.setState({checked: !this.state.checked})
        }


    }
    render() {
        return (
            <tr>
                <td style = {{ border: "1px solid black" }} >
                    <input type="checkbox"  checked = {this.state.checked} onChange = {this.handleChange.bind(this)}/>
                </td>
                <td style ={style.tableContent}>{this.props.title}</td>
                <td style={{border: "1px solid black"}} >{this.props.children}</td>
            </tr>
        );
    }
}
Todo.propTypes = {
    title: React.PropTypes.string.isRequired
};

let style = {
    tableContent: {
        border: "1px solid black"
    }
};
```

**Comment by dmwelch** [comment]
The instructions for the step are pretty confusing, especially if you aren't familiar with webpack:

> Finally, fix the router of /bundle.js and / like below.
> If you have an access to /bundle.js, you change app.js that can work on the front-end and response.
> If you have an access to /, you response HTML that consists of reading index.jsx, and data from server, and bundle.js.

It's not even valid English: "you change app.js that can work on the front-end and response" and "you response HTML".  Me Tarzan, you React.  Use verbs.

**Comment by peterrstanley** [comment]
I'm not sure if I can still reply to this to help solve or discuss any other issues, but I thought it would be better than opening a new ticket about the same thing. 

I am have the same issue with this problem. I was able to resolve one problem in regards to an error I was receiving: 

> TypeError: Cannot read property 'body' of undefined

I solved this by doing the following: 
Replace this line
`var DOM = React.DOM;`
With
`var DOM = require('react-dom');`

This resolved one issue, but brought a new error:

> TypeError: div is not a function

This error is reference to the following code:

<pre>
var html = ReactDOMServer.renderToStaticMarkup(body(null,
  div({id: 'app', dangerouslySetInnerHTML: {__html: markup}}),
  script({
    id: 'initial-data',
    type: 'text/plain',
    'data-json': initialData
  }),
  script({src: '/bundle.js'})
));
</pre>

Is there anyone that can help with this issue?
